File: occasum-core/src/main/java/pw/stamina/occasum/AbstractNamed.java
package pw.stamina.occasum;

public abstract class AbstractNamed implements Named {
    protected final String name;
    protected final String id;

    protected AbstractNamed(String name) {
        this.name = Named.validateName(name);
        this.id = Named.createIdFromName(name);
    }

    @Override
    public final String getName() {
        return name;
    }

    @Override
    public final String getId() {
        return id;
    }
}

File: occasum-core/src/main/kotlin/pw/stamina/occasum/AbstractNamed.kt
package pw.stamina.occasum

abstract class AbstractNamed protected constructor(name: String) : Named {

    final override val name: String
    final override val id: String

    init {
        Named.validateName(name)

        this.name = name
        this.id = Named.createIdFromName(name)
    }
}
